core/server/api/canary/integrations.js

```
@@ -74,6 +75,23 @@ module.exports = {
             }
         },
         query({data, options}) {
+            if (options.keyid) {
+                return models.ApiKey.findOne({id: options.keyid})
+                    .then(async (model) => {
+                        if (!model) {
+                            throw new common.errors.NotFoundError({
+                                message: common.i18n.t('errors.api.resource.resourceNotFound', {
+                                    resource: 'ApiKey'
+                                })
+                            });
+                        }
+                        await models.ApiKey.refreshSecret(model.toJSON(), Object.assign({}, options, {id: options.keyid}));
```

@kevinansfield Though ideally shouldn't happen, in case of the refresh secret failure here should we throw a generic error or something specific 🤔

---

Hmm, throwing a 500 error would probably be useful here so that we can track it properly if something does go wrong. Don't think there's any need for something specific for the client

---

@kevinansfield Updated to throw a generic `GhostError` in case of any failures in refresh now